src/main/java/reposense/parser/CsvParser.java

```diff
@@ -211,22 +220,45 @@ private String getRowContentAsRawString(final CSVRecord record) {
     }
 
     /**
-     * Checks if {@code possibleHeader} contains the expected number of columns.
-     * @throws InvalidCsvException if {@code possibleHeader} does not have as many columns as expected.
+     * Generates map of column header to position number for input {@code possibleHeader}.
+     * @throws InvalidCsvException if {@code possibleHeader} does not contain all the mandatory headers.
      */
     private void validateHeader(String[] possibleHeader) throws InvalidCsvException {
-        int actualNumberOfColumns = possibleHeader.length;
-        if (actualNumberOfColumns != expectedHeaderSize) {
-            throw new InvalidCsvException(String.format(
-                    MESSAGE_WRONG_HEADER_SIZE, csvFilePath.getFileName(), actualNumberOfColumns,
-                    expectedHeaderSize));
+        headerSize = possibleHeader.length;
+        for (int i = 0; i < headerSize; i++) {
+            String possible = possibleHeader[i].trim();
+            for (String parsedHeader : mandatoryHeaders()) {
+                if (possible.equalsIgnoreCase(parsedHeader)) {
+                    headerMap.put(parsedHeader, i);
+                    break;
```

Comment:
Why is the 'break' been added here? Is it not supposed to be checking for all the mandatory headers?

Reply:
The break is there because I am assuming that the parsed column header would match with at most 1 of the mandatory headers, i.e. that we would not have a case where there are 2 mandatory headers which are identical.

Reply:
Then, it seems that in a certain scenario, duplicated column names cannot be found out?

Reply:
> 
> 
> Then, it seems that in a certain scenario, duplicated column names cannot be found out?

I have added a warning if there are duplicate headers detected in the CSV.